editor/blocks/list/index.js

```diff
@@ -53,29 +57,57 @@ wp.blocks.registerBlock( 'core/list', {
 			onClick( attributes, setAttributes ) {
 				setAttributes( { align: 'justify' } );
 			}
+		},
+		{
+			icon: 'editor-ul',
+			title: wp.i18n.__( 'Convert to unordered' ),
+			isActive: ( { } ) => false, // To be implemented
+			onClick( attributes, setAttributes ) {
+				setAttributes( { listType: 'ul' } );
+			}
+		},
+		{
+			icon: 'editor-ol',
+			title: wp.i18n.__( 'Convert to ordered' ),
+			isActive: ( { } ) => false, // To be implemented
+			onClick( attributes, setAttributes ) {
+				setAttributes( { listType: 'ol' } );
+			}
 		}
+
 	],
 
-	edit( { attributes } ) {
+	edit( { attributes, setAttributes } ) {
 		const { listType = 'ol', items = [], align } = attributes;
 		const content = items.map( item => {
 			return `<li>${ item.value }</li>`;
 		} ).join( '' );
-
 		return (
 			<Editable
 				tagName={ listType }
 				style={ align ? { textAlign: align } : null }
+				onChange={ ( v, node ) => {
+					// The node is getting passed through from Tiny.
+					// Is there a way to parse this with hpq that doesn't
+					// involve a string?
+					const attrs = parse( '<' + node.nodeName + '>' + v + '</' + node.nodeName + '>', parseAttrs );
```

Comment:
I think the concise hpq parse approach would be cleaner. It could be reused from the previous attributes declaration also.